actors/builtin/miner/miner_state_test.go

```diff
@@ -398,6 +398,155 @@ func TestPostSubmissionsBitfield(t *testing.T) {
 	})
 }
 
+func TestVesting_AddLockedFunds_Table(t *testing.T) {
+	vestStart := abi.ChainEpoch(10)
+	vestSum := int64(100)
+
+	testcase := []struct {
+		desc   string
+		vspec  *miner.VestSpec
+		vepocs []int64
+	}{
+		{
+			desc: "vest funds in a single epoch",
+			vspec: &miner.VestSpec{
+				InitialDelay: 0,
+				VestPeriod:   1,
+				StepDuration: 1,
+				Quantization: 1,
+			},
+			vepocs: []int64{0, 0, 100, 0},
+		},
+		{
+			desc: "vest funds with period=2",
+			vspec: &miner.VestSpec{
+				InitialDelay: 0,
+				VestPeriod:   2,
+				StepDuration: 1,
+				Quantization: 1,
+			},
+			vepocs: []int64{0, 0, 50, 50, 0},
+		},
+		{
+			desc: "vest funds with period=2 quantization=2",
+			vspec: &miner.VestSpec{
+				InitialDelay: 0,
+				VestPeriod:   2,
+				StepDuration: 1,
+				Quantization: 2,
+			},
+			vepocs: []int64{0, 0, 0, 100, 0},
+		},
+		{desc: "vest funds with period=3",
+			vspec: &miner.VestSpec{
+				InitialDelay: 0,
+				VestPeriod:   3,
+				StepDuration: 1,
+				Quantization: 1,
+			},
+			vepocs: []int64{0, 0, 33, 33, 34, 0},
+		},
+		{
+			desc: "vest funds with period=3 quantization=2",
+			vspec: &miner.VestSpec{
+				InitialDelay: 0,
+				VestPeriod:   3,
+				StepDuration: 1,
+				Quantization: 2,
+			},
+			vepocs: []int64{0, 0, 0, 66, 0, 34, 0},
+		},
+		{desc: "vest funds with period=2 step=2",
+			vspec: &miner.VestSpec{
+				InitialDelay: 0,
+				VestPeriod:   2,
+				StepDuration: 2,
+				Quantization: 1,
+			},
+			vepocs: []int64{0, 0, 0, 100, 0},
+		},
+		{
+			desc: "vest funds with period=5 step=2",
+			vspec: &miner.VestSpec{
+				InitialDelay: 0,
+				VestPeriod:   5,
+				StepDuration: 2,
+				Quantization: 1,
+			},
+			vepocs: []int64{0, 0, 0, 40, 0, 40, 0, 20, 0},
+		},
+		{
+			desc: "vest funds with period=5 step=2 quantization=2",
+			vspec: &miner.VestSpec{
+				InitialDelay: 0,
+				VestPeriod:   5,
+				StepDuration: 2,
+				Quantization: 2,
+			},
+			vepocs: []int64{0, 0, 0, 40, 0, 40, 0, 20, 0},
+		},
+		{
+			desc: "vest funds with period=5 step=3 quantization=1",
+			vspec: &miner.VestSpec{
+				InitialDelay: 0,
+				VestPeriod:   5,
+				StepDuration: 3,
+				Quantization: 1,
+			},
+			vepocs: []int64{0, 0, 0, 0, 60, 0, 0, 40, 0},
+		},
+		{
+			desc: "vest funds with period=5 step=3 quantization=2",
+			vspec: &miner.VestSpec{
+				InitialDelay: 0,
+				VestPeriod:   5,
+				StepDuration: 3,
+				Quantization: 2,
+			},
+			vepocs: []int64{0, 0, 0, 0, 0, 80, 0, 20, 0},
+		},
+		{
+			desc: "(step greater than period) vest funds with period=5 step=6 quantization=1",
+			vspec: &miner.VestSpec{
+				InitialDelay: 0,
+				VestPeriod:   5,
+				StepDuration: 6,
+				Quantization: 1,
+			},
+			vepocs: []int64{0, 0, 0, 0, 0, 0, 0, 100, 0},
+		},
+		{
+			desc: "vest funds with delay=5 period=5 step=1 quantization=1",
+			vspec: &miner.VestSpec{
+				InitialDelay: 5,
+				VestPeriod:   5,
+				StepDuration: 1,
+				Quantization: 1,
+			},
+			vepocs: []int64{0, 0, 20, 20, 20, 20, 20, 0},
```

Comment:
This one looks weird because it looks like the delay is being ignored. I suggest removing the  `+tc.vspec.InitialDelay` on line 538, and putting in the leading zeros in this result.

Alternatively break out a new set of tests for non-zero delays.

Reply:
good idea, adjusted in latest commit.